### ISSUE TITLE
[Accessibility][JAWS][Select]: disabled select is not announced

### ISSUE THREAD

**Issue by silvalaura** [cengage/react-magma#2181]
[A11Y-4169](https://jira.cengage.com/browse/A11Y-4169)

_reported in another team's board_

Platform:
Windows 11 / Chrome /Keyboard
Mac15/ Safari/ Voiceover

Steps to reproduce:
1. Open the URL "https://faculty-stage.cengage.com/"
2. Click on "sign in" button
3. On sign in page enter instructor email and password and do sign in
4. Select "My course" from main menu
5. Select course key: MTPQGMDZD55G
6. Navigate near "Bulk Edit" and activate it.
7. Select the content and move near "Edit X activities" button.
8. Expand the button, and select "edit student accomodations" and expand it.
9. Navigate to a form or UI containing a `<select>` dropdown with one or more disabled options.
10. Observe that “Unavailable” status is not announced by screen readers for a select option when the option is disabled or not selectable.

Actual Result:
The screen reader reads only the visible text of the disabled option but does not indicate that it is unavailable or disabled.

Expected Result:
Screen readers should clearly announce the disabled/unavailable status of the option (e.g., "Option name, unavailable" or "Option name, disabled").

WCAG Guideline:
4.1.2: Name, Role, Value

User Impact:
Users relying on screen readers may believe an unavailable option is selectable. This can lead to confusion, frustration, and incomplete form submission or incorrect choices.

Recommendation:
Use the native disabled attribute for `<option>` elements that are not selectable in a standard HTML `<select>` element. For custom dropdown components, apply aria-disabled="true" to unavailable options and ensure the screen reader can detect and announce the state.

Screencast:
https://app.screencast.com/gsMJ0XBqRsqTE

**Team Note:** 
our Select component, when disabled, should announce to screenreader that it's disabled. It should probably behave the same as the NativeSelect

**Comment by nikitaorliak-cengage** [comment]
On MacOS + VO it announces that Select (and Multiselect) is dimmed, not disabled. On Windows can't move focus to the disabled select.

<img width="781" height="345" alt="Image" src="https://github.com/user-attachments/assets/921a5136-55f6-4482-99ad-1217ba165420" />

**Comment by moathabuhamad-cengage** [comment]
VoiceOver uses "dimmed" to describe aria-disabled="true", Moving this to done

**Comment by olehnoskov-cengage** [comment]
> We reviewed the behavior across the supported screen reader and browser combinations (Edge + NVDA, Chrome + JAWS,  > Safari + VoiceOver) and confirmed that the component is announced correctly in all supported configurations.

> Hence, we are closing the ticket.
